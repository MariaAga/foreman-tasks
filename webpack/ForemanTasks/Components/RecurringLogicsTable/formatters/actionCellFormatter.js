import React from 'react';
import { cellFormatter } from 'foremanReact/components/common/table';
import { translate as __ } from 'foremanReact/common/I18n';
import { ActionButtons } from 'foremanReact/components/common/ActionButtons/ActionButtons';

export const actionCellFormatter = () => (
  canEdit,
  { rowData: { id, state } }
) => {
  const buttons = [];
  if (canEdit) {
    if (state === 'disabled') {
      buttons.push({
        title: __('Enable'),
        action: {
          rel: 'nofollow',
          'data-method': 'put',
          href: `/foreman_tasks/recurring_logics/${id}/enable`,
          id: `enable-button-${id}`,
        },
      });
    }
    if (state !== 'cancelled' && state !== 'finished' && state !== 'disabled') {
      buttons.push({
        title: __('Disable'),
        action: {
          rel: 'nofollow',
          'data-method': 'put',
          href: `/foreman_tasks/recurring_logics/${id}/disable`,
          id: `disable-button-${id}`,
        },
      });
    }
    if (state !== 'cancelled' && state !== 'finished') {
      buttons.push({
        title: __('Cancel'),
        action: {
          rel: 'nofollow',
          'data-method': 'post',
          href: `/foreman_tasks/recurring_logics/${id}/cancel`,
          id: `cancel-button-${id}`,
        },
      });
    }
  }
  return cellFormatter(<ActionButtons buttons={buttons} />);
};
