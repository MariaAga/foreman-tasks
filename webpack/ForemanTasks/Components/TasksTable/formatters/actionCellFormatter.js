import React from 'react';
import { cellFormatter } from 'foremanReact/components/common/table';
import { ActionButton } from '../../common/ActionButtons/ActionButton';

export const actionCellFormatter = taskActions => (
  value,
  { rowData: { action, id } }
) =>
  cellFormatter(
    <ActionButton
      id={id}
      name={action}
      taskActions={taskActions}
      availableActions={value}
    />
  );
