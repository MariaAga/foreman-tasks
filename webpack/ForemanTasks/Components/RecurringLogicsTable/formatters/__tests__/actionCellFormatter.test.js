import { actionCellFormatter } from '../actionCellFormatter';

describe('actionCellFormatter', () => {
  it('render disabled', () => {
    const data = [true, { rowData: { id: 1, state: 'disabled' } }];
    expect(actionCellFormatter({})(...data)).toMatchSnapshot();
  });
  it('render active', () => {
    const data = [true, { rowData: { id: 1, state: 'active' } }];
    expect(actionCellFormatter({})(...data)).toMatchSnapshot();
  });
  it('render finished', () => {
    const data = [true, { rowData: { id: 1, state: 'finished' } }];
    expect(actionCellFormatter({})(...data)).toMatchSnapshot();
  });
  it('render cancelled', () => {
    const data = [true, { rowData: { id: 1, state: 'cancelled' } }];
    expect(actionCellFormatter({})(...data)).toMatchSnapshot();
  });
  it('render no edit', () => {
    const data = [false, { rowData: {} }];
    expect(actionCellFormatter({})(...data)).toMatchSnapshot();
  });
});
