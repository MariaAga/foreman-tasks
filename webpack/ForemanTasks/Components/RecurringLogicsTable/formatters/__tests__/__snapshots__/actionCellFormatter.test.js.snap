// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionCellFormatter render active 1`] = `
<ActionButtons
  buttons={
    Array [
      Object {
        "action": Object {
          "data-method": "put",
          "href": "/foreman_tasks/recurring_logics/1/disable",
          "id": "disable-button-1",
          "rel": "nofollow",
        },
        "title": "Disable",
      },
      Object {
        "action": Object {
          "data-method": "post",
          "href": "/foreman_tasks/recurring_logics/1/cancel",
          "id": "cancel-button-1",
          "rel": "nofollow",
        },
        "title": "Cancel",
      },
    ]
  }
/>
`;

exports[`actionCellFormatter render cancelled 1`] = `
<ActionButtons
  buttons={Array []}
/>
`;

exports[`actionCellFormatter render disabled 1`] = `
<ActionButtons
  buttons={
    Array [
      Object {
        "action": Object {
          "data-method": "put",
          "href": "/foreman_tasks/recurring_logics/1/enable",
          "id": "enable-button-1",
          "rel": "nofollow",
        },
        "title": "Enable",
      },
      Object {
        "action": Object {
          "data-method": "post",
          "href": "/foreman_tasks/recurring_logics/1/cancel",
          "id": "cancel-button-1",
          "rel": "nofollow",
        },
        "title": "Cancel",
      },
    ]
  }
/>
`;

exports[`actionCellFormatter render finished 1`] = `
<ActionButtons
  buttons={Array []}
/>
`;

exports[`actionCellFormatter render no edit 1`] = `
<ActionButtons
  buttons={Array []}
/>
`;
